### PR TITLE
(FACT-2786) Fix fact caching if fact is defined in multiple groups

### PR DESCRIPTION
**Pull request by florindragos**
This PR also enables fact caching by only specifying individual facts in ttls configuration, rather than needing a cache group.

### CODE REVIEW

**Comment by florindragos** [comment]
Jenkins please test this on all

File: lib/facter/framework/config/fact_groups.rb [hunk 1]
```diff
@@ -41,6 +45,15 @@ def get_group_ttls(group_name)
       ttls_to_seconds(ttls[group_name])
     end
 
+    def get_fact(fact_name)
+      return @facts_ttls[fact_name] if @facts_ttls[fact_name]
+
+      fact = @facts_ttls.select { |k, v| break v if fact_name =~ /^#{k}\..*/ }
```
Comment (gimmyxd) [inline lib/facter/framework/config/fact_groups.rb:51]: `k, v` are not very suggestive, maybe use `key`, `value`.
Also that break there is a bit hard to read overall. Do you need just the first value that matches `/^#{k}\..*/`?

**Comment by florindragos** [comment]
Jenkins please test this on all

**Comment by florindragos** [comment]
Jenkins please test this on all

File: lib/facter/framework/config/fact_groups.rb (hunk 1 above)
Comment (florindragos) [reply lib/facter/framework/config/fact_groups.rb]: Yes, we take the first match. `@facts_ttls` is sorted so the children come first (os.macosx vs os)

**Comment by florindragos** [comment]
Jenkins please test this on all

File: lib/facter/framework/config/fact_groups.rb (hunk 1 above)
Comment (Filipovici-Andrei) [reply lib/facter/framework/config/fact_groups.rb]: You could use detect instead of select. It returns the first element for which the block returns true.

**Comment by Filipovici-Andrei** [comment]
Is there a unit test that checks if a fact contained in two groups is not cached twice?

**Comment by florindragos** [comment]
> Is there a unit test that checks if a fact contained in two groups is not cached twice?

too complex for a unit test. added acceptance test.

**Review by Filipovici-Andrei — APPROVED** [review]

**Pull request merged**